guides/source/initialization.md

```diff
@@ -1,10 +1,610 @@
+
+So now we know what role Railties and Engines play in the Rails initialization process. Back to the the `initialize!` method that is called from `config/environment.rb`.
+
+### The `initialize!` call
+
+The `initialize!` method looks like this:
+
+```ruby
+# railties/lib/rails/application.rb
+
+def initialize!(group = :default)
+  raise "Application has been already initialized." if @initialized
+  run_initializers(group, self)
+  @initialized = true
+  self
+end
+```
+
+The `run_initializers` method looks like this:
+
+```ruby
+# railties/lib/rails/initializable.rb
+
+def run_initializers(group = :default, *args)
+  return if instance_variable_defined?(:@ran)
+  initializers.tsort_each do |initializer|
+    initializer.run(*args) if initializer.belongs_to?(group)
+  end
+  @ran = true
+end
+```
+
+The `run_initializers` method runs all of the initializers registered by the Engine and Railtie classes, as well as other groups of initializers (as defined in `Boostrap` and `Finisher`):
+
+```ruby
+# railties/lib/rails/application.rb
+
+def initializers
+	Bootstrap.initializers_for(self) +
+	railties_initializers(super) +
+	Finisher.initializers_for(self)
+end
+```
+
+The `bootstrap` initializers prepare the application (like initializing the
+logger) while the `finisher` initializers (like building the middleware stack)
+are run last. The `railtie` initializers are the initializers which have been
+defined on the `Rails::Application` itself and are run between the `bootstrap`
+and `finisher`.
+
+The [custom initializers](configuring.html#using-initializer-files) in your application under `config/initializers` are also run as part of the `initialize!` call. They are registered via [`load_config_initializer`](https://github.com/rails/rails/blob/36bd50c82b46046c9f352e06fa552221586028d8/railties/lib/rails/engine.rb#L643) in `Rails::Engine` Railtie. They are run in alphabetical order by file names in that directory.
+
+At the end of the `Rails.application.initialize!` call, the Rails framework and application are loaded and initialized! We're also at the end of the `config/environment.rb` file.
+
+Now all that is left to do is to start the server. Before that, let's see how you can add custom code to the initialization process.
+
+Hooking Into the Initialization Process
+---------------------------------------
+
+### Lazy Load Hooks
+
+The purpose of [Lazy Load Hooks](https://api.rubyonrails.org/v8.0/classes/ActiveSupport/LazyLoadHooks.html) is to do something when the application loads certain parts of the Rails framework. For example:
+
+```ruby
+# config/initializers/my_active_record_extension.rb
+ActiveSupport.on_load(:active_record) do
+  include MyActiveRecordExtension
+end
+```
+
+The Rails framework is responsible for loading the components and when a specific component (e.g. Active Record above) is done loading, you can choose to be notified with the `ActiveSupport.on_load` helper.
```

Comment:
While the load hooks can be seen as a notification that a component has loaded I think it's more about executing a piece of code:
```suggestion
The Rails framework is responsible for loading the components and when a specific component (e.g. Active Record above) is done loading, you can choose to execute code with the `ActiveSupport.on_load` helper.
```